### ISSUE TITLE
Any good ways to get access to the output shape of certain previous layer?

### ISSUE THREAD

**Issue by ypxie** [fchollet/keras#1981]
Need this information to either pad or crop the feature map to train the fully convolutional neural network.

**Comment by jliemansifry** [comment]
Yes! Build your model, access the layers with model.layers, then check out the attributes of your layer of interest. `.input_shape and .output_shape` are what you're looking for.

**Comment by shampool** [comment]
Thx, But I need that information to be included in the computation graph. Namely, the running computation needs to adjust the computation based on that.

**Comment by Edouard360** [comment]
Indeed, `model.layers` is only accessible once you have instantiated your model.
Using the `._keras_shape` attribute of the output of any instantiated `Layer`, you should be able to get the output_shape of your tensor and run your computations accordingly. For instance:

`x = keras.layers.Input(x_train.shape[1:])`
`conv1 = keras.layers.Conv1D(16, 8, padding='same')(x)`
`print(conv1._keras_shape)`

I think if you want to be "cleaner" you should create a custom layer (https://keras.io/layers/writing-your-own-keras-layers/) and use the `input_shape`argument in the `build` function to get the relevant information that you need. This is for the previous layer.

If you want information from all the previous layers shapes, you can use `get_input_shape_at(node_index)`, which is an inherited function from the `Layer` class.

**Comment by emerygoossens** [comment]
Simply call the output or output.shape 

model.layers[5].output

or 

model.layers[5].output.shape
